src/dotnet-install.sh

```diff
@@ -1469,28 +1501,45 @@ say "- The SDK needs to be installed without user interaction and without admin
 say "- The SDK installation doesn't need to persist across multiple CI runs."
 say "To set up a development environment or to run apps, use installers rather than this script. Visit https://dotnet.microsoft.com/download to get the installer.\n"
 
+if [ "$internal" = true ] && [ -z "$feed_credential" ]; then
+    message="Provide credentials via --feed-credential parameter."
+    if [ "$dry_run" = true ]; then
```

Comment:
Well, it is less code if we always fail, right, but to me --dry-run should not fail in this case, as it really does not need the access to the resource.

Reply:
DryRun still downloads files to determine the latest version of that channel (when used with channels that don't support aka.ms links). They will see a 404 error in that case because the credentials are not there. I think it is better to prevent that scenario.